exchange/exchange.go

```diff
@@ -75,6 +76,10 @@ type bidResponseWrapper struct {
 func NewExchange(client *http.Client, cache prebid_cache_client.Client, cfg *config.Configuration, metricsEngine pbsmetrics.MetricsEngine, infos adapters.BidderInfos, gDPR gdpr.Permissions, currencyConverter *currencies.RateConverter) Exchange {
 	e := new(exchange)
 
+	var s struct{}
+	for _, c := range cfg.GDPR.EEACountries {
+		e.eeaCountries[c] = s
+	}
```

Comment:
inside `config/config.go` we build the `NonStandardPublisherMap` and `BlacklistedAppMap` in a similar fashion, maybe we can refactor in the future to set up the map in the config file instead.
```
153   type GDPR struct {
154       HostVendorID            int          `mapstructure:"host_vendor_id"`
155       UsersyncIfAmbiguous     bool         `mapstructure:"usersync_if_ambiguous"`
156       Timeouts                GDPRTimeouts `mapstructure:"timeouts_ms"`
157       NonStandardPublishers   []string     `mapstructure:"non_standard_publishers,flow"`
158       NonStandardPublisherMap map[string]int
159       TCF1                    TCF1 `mapstructure:"tcf1"`
160       TCF2                    TCF2 `mapstructure:"tcf2"`
161       AMPException            bool `mapstructure:"amp_exception"`
162       // EEACountries (EEA = European Economic Area) are a list of countries where we should assume GDPR applies.
163       // If the gdpr flag is unset in a request, but geo.country is set, we will assume GDPR applies if and only
164       // if the country matches one on this list. If both the GDPR flag and country are not set, we default
165       // to UsersyncIfAmbiguous
166       EEACountries []string `mapstructure:"eea_countries"`
    +     EEACountryMap map[string]struct{}
167   }
168   *--360 lines: func (cfg *GDPR) validate(errs configErrors) configErrors {------------------------------------
528   // New uses viper to get our server configurations.
529   func New(v *viper.Viper) (*Configuration, error) {
530       var c Configuration
531       if err := v.Unmarshal(&c); err != nil {
532           return nil, fmt.Errorf("viper failed to unmarshal app config: %v", err)
533       }
534       c.setDerivedDefaults()
535  
536   *--  8 lines: if err := c.RequestValidation.Parse(); err != nil {--------------------------------------------
544  
545       // To look for a request's publisher_id in the NonStandardPublishers list in
546       // O(1) time, we fill this hash table located in the NonStandardPublisherMap field of GDPR
547       c.GDPR.NonStandardPublisherMap = make(map[string]int)
548       for i := 0; i < len(c.GDPR.NonStandardPublishers); i++ {
549           c.GDPR.NonStandardPublisherMap[c.GDPR.NonStandardPublishers[i]] = 1
550       }
551  
    +     var s struct{}
    +     for _, c := range c.GDPR.EEACountries {
    +     	c.GDPR.EEACountryMap[c] = s
    +     }
552       // To look for a request's app_id in O(1) time, we fill this hash table located in the
553       // the BlacklistedApps field of the Configuration struct defined in this file
554       c.BlacklistedAppMap = make(map[string]bool)
555       for i := 0; i < len(c.BlacklistedApps); i++ {
556           c.BlacklistedAppMap[c.BlacklistedApps[i]] = true
557       }
config/config.go
```

Reply:
It's only used in the exchange endpoint though. I'm personally happy with its current location.